Remove compiled files from version control and add to .gitignore

BerriJ opened the pull request:
This PR removes some `.o`, `.so`, and `,dll` files from git version control and adds them to the .gitignore file.

Almost certainly, these need to be compiled locally during package install or other functions that start compilation of the respective `C` code.

At the moment:
1. Cloning the repository
2. Executing `devtools::load_all()` 

Leads to an error message due to the presence of these files (devtools assumes that it does not need to compile again):

```
[R] devtools::load_all()
ℹ Loading gamlss
Loading required package: splines
Loading required package: gamlss.data

Attaching package: ‘gamlss.data’

The following object is masked from ‘package:datasets’:

    sleep

Loading required package: gamlss.dist
Loading required package: nlme
Loading required package: parallel
Error in dyn.load(dll_copy_file) : 
  unable to load shared object '/tmp/RtmpyAVSHy/pkgloadef057eaaa2/gamlss.so':
  /tmp/RtmpyAVSHy/pkgloadef057eaaa2/gamlss.so: invalid ELF header
```

Merging this PR solves this. A call to `devtools::load_all()` will trigger compilation as it should:

```
[R] devtools::load_all()
ℹ Loading gamlss
ℹ Re-compiling gamlss (debug build)
── R CMD INSTALL ──────────────────────────────────────────────────────────────────
─  installing *source* package ‘gamlss’ ...
   ** using staged installation
   ** libs
   using C compiler: ‘gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0’
   ccache gcc -I"/usr/share/R/include" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-H0vbME/r-base-4.3.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -UNDEBUG -Wall -pedantic -g -O0 -fdiagnostics-color=always -c gamlss_init.c -o gamlss_init.o
   ccache gcc -I"/usr/share/R/include" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-H0vbME/r-base-4.3.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -UNDEBUG -Wall -pedantic -g -O0 -fdiagnostics-color=always -c genD.c -o genD.o
   ccache gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro -o gamlss.so gamlss_init.o genD.o -L/usr/lib/R/lib -lR
   installing to /tmp/RtmpyAVSHy/devtools_install_ef07715f8e/00LOCK-gamlss/00new/gamlss/libs
   ** checking absolute paths in shared objects and dynamic libraries
─  DONE (gamlss)
 **********   GAMLSS Version 5.4-22  ********** 
For more on GAMLSS look at https://www.gamlss.com/
Type gamlssNews() to see new features/changes/bug fixes
```

Best Regards,
BerriJ

zeileis commented:
Thanks for the PR. I agree that the targets should not be in the git repository and adding them to the `.gitignore` is a good idea. Mikis @mstasinopoulos, is there a particular reason for adding these files?

BerriJ commented:
Just a quick reminder about this pull request—no rush, but I wanted to check if you’ve had a chance to take a look. Let me know if there’s anything I can clarify or update on my end.

mstasinopoulos commented:
Dear Achim

What action I have to take? It is not clear to me 

Do I have to remove the .o .so and .dll files? 
Or just put them into a different directory? 

I am not sure I understand he PR

Thanks

Mikis




> On 9 Jun 2025, at 09:32, Jonathan Berrisch ***@***.***> wrote:
> 
> 
> BerriJ
>  left a comment 
> (gamlss-dev/gamlss#8)
>  <https://github.com/gamlss-dev/gamlss/pull/8#issuecomment-2955071435>
> Just a quick reminder about this pull request—no rush, but I wanted to check if you’ve had a chance to take a look. Let me know if there’s anything I can clarify or update on my end.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/gamlss-dev/gamlss/pull/8#issuecomment-2955071435>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AE7QN7X26HSAV5677FZMU733CVBBVAVCNFSM6AAAAAB64K3MYCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDSNJVGA3TCNBTGU>.
> You are receiving this because you were mentioned.
>

BerriJ commented:
Actually, I already did the necessary changes. You can see those here: https://github.com/gamlss-dev/gamlss/pull/8/files (or by clicking on the "Files Changed" Tab on top of this page.

Basically, I just removed the compiled files (those with .dll and .o endings) and added those to the .gitignore, so git won't add them again.

If you want to adtop these changes , you can just click the "Merge Pull Request" button. Should be a little bit below this message.

zeileis commented:
Yes, I completely agree.

_(The reason I didn't accept this PR was that I wasn't sure whether there is a reason for including the files in the first place. But this doesn't seem to be the case.)_
